scripts/generate-helm-charts.sh

```diff
@@ -0,0 +1,189 @@
+    local expose=false
+  fi
+
+  echo "Generating app resources for $project/$container_tag/$deployment_type"
+  rm -rf $project/target
+
+  $project/mvnw -f $project/pom.xml versions:set clean package \
+    -DskipTests \
+    -DnewVersion=$container_tag \
+    -Dmaven.compiler.release=$javaVersion \
+    -Dquarkus.container-image.tag=$container_tag \
+    -Dquarkus.kubernetes.deployment-target=$deployment_type \
+    -Dquarkus.kubernetes.version=$container_tag \
+    -Dquarkus.kubernetes.ingress.expose=$expose \
+    -Dquarkus.kubernetes.resources.limits.memory=$mem_limit \
+    -Dquarkus.kubernetes.resources.requests.memory=$mem_request \
+    -Dquarkus.kubernetes.annotations.\"app.quarkus.io/vcs-url\"=$GITHUB_SERVER_URL/$GITHUB_REPOSITORY \
+    -Dquarkus.kubernetes.annotations.\"app.quarkus.io/vcs-ref\"=$github_ref_name \
+    -Dquarkus.openshift.version=$container_tag \
+    -Dquarkus.openshift.route.expose=$expose \
+    -Dquarkus.openshift.resources.limits.memory=$mem_limit \
+    -Dquarkus.openshift.resources.requests.memory=$mem_request \
+    -Dquarkus.openshift.annotations.\"app.openshift.io/vcs-url\"=$GITHUB_SERVER_URL/$GITHUB_REPOSITORY \
+    -Dquarkus.openshift.annotations.\"app.openshift.io/vcs-ref\"=$github_ref_name \
+    -Dquarkus.knative.version=$container_tag \
+    -Dquarkus.knative.labels.\"app.openshift.io/runtime\"=quarkus \
+    -Dquarkus.knative.resources.limits.memory=$mem_limit \
+    -Dquarkus.knative.resources.requests.memory=$mem_request \
+    -Dquarkus.knative.annotations.\"app.openshift.io/vcs-url\"=$GITHUB_SERVER_URL/$GITHUB_REPOSITORY \
+    -Dquarkus.knative.annotations.\"app.openshift.io/vcs-ref\"=$github_ref_name \
+    -Dquarkus.helm.version=1.0.0 \
+    -Dquarkus.helm.name=$project-$container_tag-$deployment_type
+}
+
+process_quarkus_project() {
+  local project=$1
+  local deployment_type=$2
+  local version_tag=$3
+  local javaVersion=$4
+  local kind=$5
+  local container_tag="${version_tag}-latest"
+  local output_filename="${version_tag}-${deployment_type}"
+  local app_generated_input_file="$project/target/kubernetes/${deployment_type}.yml"
+  local project_output_file="$project/$OUTPUT_DIR/${output_filename}.yml"
+  local all_apps_output_file="$OUTPUT_DIR/${output_filename}.yml"
+
+  local app_generated_helm_chart="$project/target/helm/${deployment_type}/${project}-${container_tag}-${deployment_type}"
+  local generated_helm_dir="$project/${OUTPUT_HELM_DIR}/${deployment_type}"
+
+  # 1st do the build
+  # The build will generate all the resources for the project
+  do_build $project $deployment_type $version_tag $javaVersion $kind
+
+  rm -rf $generated_helm_dir
+  mkdir $generated_helm_dir
+
+  # Now merge the generated resources to the top level (deploy/helm)
+  if [[ -f "$app_generated_input_file" ]]; then
+    echo "Copying app generated helm ($app_generated_input_file) to $project_output_file and $all_apps_output_file"
+
+    cp -R $app_generated_helm_chart $generated_helm_dir
+
+  fi
```

Comment:
The folder structure here deviates from other deployment types.
P.e. for k8s we have `k8s/{version-tag}-{deployment-type}`, while for helm we have `helm/{deployment-type}/{chart-name}`.

Proposal: Aligning a bit, generating `helm/{version-tag}/{deployment-type}`:
```suggestion
  local app_generated_helm_chart="$project/target/helm/${deployment_type}/${project}-${container_tag}-${deployment_type}"
  local generated_helm_dir="$project/${OUTPUT_HELM_DIR}/${version_tag}/${deployment_type}"

  # 1st do the build
  # The build will generate all the resources for the project
  do_build $project $deployment_type $version_tag $javaVersion $kind

  rm -rf $generated_helm_dir
  mkdir -p $generated_helm_dir

  # Now copy the helm files into the deploy directory (deploy/helm) out of the transient target.
  if [[ -f "$app_generated_input_file" ]]; then
    echo "Copying generated helm chart ($app_generated_helm_chart) to $generated_helm_dir"

    cp -R $app_generated_helm_chart/* $generated_helm_dir

  fi
```